roqoqo-iqm/src/backend.rs

```diff
@@ -451,6 +469,47 @@ impl Backend {
             msg: format!("Job did not finish in {} seconds", TIMEOUT_SECS),
         })
     }
+
+    /// Abort a submitted job.
+    ///
+    /// # Arguments
+    ///
+    /// `id` - The ID of the job to abort
+    ///
+    /// # Returns
+    ///
+    /// Err(RoqoqoBackendError) - If the job abortion failed.
+    pub fn abort_job(&self, id: &str) -> Result<(), RoqoqoBackendError> {
+        let client = reqwest::blocking::Client::builder()
+            .https_only(true)
+            .build()
+            .map_err(|x| RoqoqoBackendError::NetworkError {
+                msg: format!("could not create https client {:?}", x),
+            })?;
+
+        let abort_url = [&self.device.remote_host(), "jobs", id, "abort"].join("/");
+
+        let resp = client
+            .post(abort_url)
+            .headers(_construct_headers(&self.access_token))
+            .send()
+            .map_err(|e| RoqoqoBackendError::NetworkError {
+                msg: format!("Error during POST request for abort_job: {:?}", e),
+            })?;
+
+        match resp.status() {
+            reqwest::StatusCode::OK => Ok(()),
+            _ => {
+                let abort_failed_msg: &str =
+                    &serde_json::from_str::<AbortResponse>(&resp.text().unwrap())
+                        .unwrap()
+                        .detail;
+                Err(RoqoqoBackendError::GenericError {
+                    msg: format!("Job abortion failed: {}", abort_failed_msg),
+                })
+            }
+        }
+    }
```

Comment:
Should there be an additional test to cover this functionality?

Reply:
Since it's the live API it's hard to test that in CI.